### PR TITLE
[TwigBundle] Deprecate the public "twig" service to private

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | -
| License       | MIT
| Doc PR        | -

I think the `twig` service don't need to be public anymore - we never need to access it directly in Symfony's code.

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/FormLoginBundle.php [hunk 1]
```diff
@@ -11,8 +11,28 @@
...
+use Symfony\Bundle\SecurityBundle\Tests\Functional\Bundle\FormLoginBundle\Controller\LoginController;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
 class FormLoginBundle extends Bundle
 {
+    /**
+     * {@inheritdoc}
+     */
+    public function build(ContainerBuilder $container)
```
Comment (fancyweb) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/FormLoginBundle.php:24]: Since this test bundle is used with different configurations, it's easier to register those controllers as services here (it avoids duplication in config files).

**Review by yceruto — APPROVED** [review]

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/Controller/LoginController.php [hunk 2]
```diff
@@ -11,17 +11,23 @@
...
 
-class LoginController implements ContainerAwareInterface
+class LoginController implements ServiceSubscriberInterface
 {
-    use ContainerAwareTrait;
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
```
Comment (Tobion) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/Controller/LoginController.php:29]: why not inject dependencies explicitly?

Comment (fancyweb) [reply src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/Controller/LoginController.php:29]: I tried to modify existing tests the least possible. I mean I tried to keep `$container->get`. I guess we can inject the dependencies as well.

File: src/Symfony/Bundle/FrameworkBundle/Tests/Functional/BundlePathsTest.php [hunk 3]
```diff
@@ -40,7 +40,7 @@ public function testBundlePublicDir()
     public function testBundleTwigTemplatesDir()
     {
         static::bootKernel(['test_case' => 'BundlePaths']);
-        $twig = static::$container->get('twig');
+        $twig = static::$container->get('twig.alias');
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Functional/BundlePathsTest.php:43]: can't we get private services in tests with the special test container? why is this needed?

Comment (fancyweb) [reply src/Symfony/Bundle/FrameworkBundle/Tests/Functional/BundlePathsTest.php:43]: The `twig` service is still public. We don't want to access to it directly to avoid the deprecation.

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/Controller/LoginController.php (hunk 2 above)
Comment (fancyweb) [reply src/Symfony/Bundle/SecurityBundle/Tests/Functional/Bundle/FormLoginBundle/Controller/LoginController.php:29]: Using the service subscriber system looks OK to me 🤷‍♂️

**Review by fabpot — APPROVED** [review]
Can you rebase?

**Comment by fancyweb** [comment]
@fabpot Done.

**Review by chalasr — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.
